perllib/FixMyStreet/App/Controller/Admin/Waste.pm

```diff
@@ -0,0 +1,112 @@
+package FixMyStreet::App::Controller::Admin::Waste;
+
+use JSON::MaybeXS;
+use Moose;
+use Try::Tiny;
+use namespace::autoclean;
+
+BEGIN { extends 'Catalyst::Controller'; }
+
+=head1 NAME
+
+FixMyStreet::App::Controller::Admin::Waste - Catalyst Controller
+
+=head1 DESCRIPTION
+
+Admin pages for configuring WasteWorks parameters
+
+=head1 METHODS
+
+=cut
+
+sub index : Path : Args(0) {
+    my ( $self, $c ) = @_;
+
+    my $user = $c->user;
+
+    if ($user->is_superuser) {
+        $c->forward('fetch_wasteworks_bodies');
+    } elsif ( $user->from_body ) {
+        $c->forward('load_wasteworks_body', [ $user->from_body->id ]);
+        $c->res->redirect( $c->uri_for_action( 'admin/waste/edit', $c->stash->{body}->id ) );
+    } else {
+        $c->detach( '/page_error_404_not_found', [] );
+    }
+}
+
+sub edit : Path : Args(1) {
+    my ( $self, $c, $body_id ) = @_;
+
+    foreach (qw(status_message)) {
```

Comment:
They are synonyms, yes, but I think I prefer to use `for` for old-style C loops, rather than confusing the two types of loop together. The FMS codebase has 13 uses of `for` C-loop style, 17 uses of `for` list-loop style, and 281 uses of `foreach` so foreach would be more consistent with majority usage, at least, but no biggie.